### ISSUE TITLE
cy.wait is not waiting if in url exists *

### ISSUE THREAD

**Issue by kamiloski** [cypress-io/cypress#9305]
### Current behavior
Till v6 `cy.wait()` was waiting for route eg 

```js
cy.route('GET','api/clientid/*/accountId/*/data' ).as('history')
```
or 
```js
cy.route('GET','user/**' ).as('users')
```
But now `cy.wait` in not waiting, after remove `*` if exists at the end of URL, then works but with situation like in first example 'api/clientid/*/accountId/*/data', it's not working at all, I tried add multiple `*` remove it and not working
### Desired behavior
`cy.wait()` should wait for eg `api/clientid/*/accountId/*/data`
### Test code to reproduce

```js
cy.route('GET','user/*' ).as('users')
```

### Versions

6.0.0

**Comment by jennifer-shehane** [comment]
- What version was this previously working on?
- Can you provide the url this is meant to match? (Feel free to remove the baseUrl)
- What error are you seeing now when running this test code.

**Comment by kamiloski** [comment]
> * What version was this previously working on?
The last one, I update cypress after each message about new version
> * Can you provide the url this is meant to match? (Feel free to remove the baseUrl)
In main post there are URLs eg. `api/clientid/*/accountId/*/data` 
> * What error are you seeing now when running this test code.
Ther is just info that was waiting but not occurs but it does, and it works before because I often use * in cy.route urls

**Comment by ipinzari** [comment]
This is a big blocker for us as well. Any URL that contains a regex match such as * `/api/something/*/not/working` would not be identified as an api call.

**Comment by bahmutov** [comment]
Since `cy.route` is deprecated, could you try [cy.intercept](https://on.cypress.io/intercept) please?

**Comment by ipinzari** [comment]
I tried to migrate to ` cy.intercept `and this is where the issue is happening. cy.route works as expected.

**Comment by bahmutov** [comment]
Ok, then @ipinzari can you open a separate issue with a fully reproducible example so we can see what is going on, please. Also please open a new issue, since it might be different from this original issue by @kamiloski

**Comment by Pirlouit** [comment]
Hello,

I'm experiencing the same issue. Since Cypress v6.0, the urls with wildcards are not matched anymore.
Exemple:
The url I'd like to match: `/api/client/{clientId}/informations`
So before v6.0 I was defining my route that way:
```ts
cy.route("/api/client/*/informations").as("clientInfo");
```
Now with `intercept`: 
```ts
cy.intercept("/api/client/*/informations").as("clientInfo");
```
But with intercept, it does not work.

If I have a request on this url: `/api/client/561457/informations`
`cy.route` find it and I'm able to wait for it while `cy.intersect` does not.

Personally, to make this work, I had to change all my url patterns to RegEx.

In this case scenario, I did this:
```ts
cy.intercept(/\/api\/client\/.*\/informations/g).as("clientInfo");
```
And now I'm able to wait or stub this route.
